src/renderers/gl/renderer.cpp

```diff
@@ -252,6 +289,166 @@ class mrg::Renderer::ProgramFactory : public mir::graphics::gl::ProgramFactory
     std::mutex compilation_mutex;
 };
 
+class mrg::Renderer::OutputFilterShader
```

Comment:
Hm. I think this *could be* a [`gl::OutputSurface`](https://github.com/canonical/mir/blob/706fd3b0b8ded81274600d261b0544244d6ab816/include/platform/mir/renderer/gl/gl_surface.h#L31)?

You'd take the underlying `gl::OutputSurface` in the constructor, delegate `{make,release}_current` to it, `bind()` would make *this* framebuffer the draw framebuffer, and then `commit` would `bind()` the underlying `gl::OutputSurface`, do what `render()` does currently, and then call `commit` on the underlying surface?

That would keep all the rest of the renderer code the same, with the only difference being the content of `output_surface`?

Reply:
Changed to nesting an OutputSurface.